File: scan_cyrillic.js
const fs = require('fs');
const path = require('path');

const cyr = /[\u0400-\u04FF]/;
const base = 'src/main/java';
const files = [];
let total = 0;

function walk(dir) {
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    const p = path.join(dir, entry.name);
    if (entry.isDirectory()) walk(p);
    else if (entry.name.endsWith('.java')) {
      let txt;
      try { txt = fs.readFileSync(p, 'utf8'); } catch (e) { continue; }
      const lines = txt.split('\n');
      const n = lines.filter(l => cyr.test(l)).length;
      if (n > 0) { files.push({ n, p }); total += n; }
    }
  }
}
walk(base);
files.sort((a, b) => b.n - a.n);
console.log('files_with_cyrillic:', files.length);
console.log('cyrillic_lines_total:', total);
console.log('--- all files sorted by count ---');
for (const f of files) console.log(String(f.n).padStart(6), f.p);

File: scan_cyrillic.py
import os, re, sys

cyr = re.compile(r'[А-Яа-яЁё]')
base = 'src/main/java'
files = []
total_lines = 0
for root, dirs, fs in os.walk(base):
    for f in fs:
        if not f.endswith('.java'):
            continue
        p = os.path.join(root, f)
        with open(p, encoding='utf-8', errors='replace') as fh:
            lines = fh.readlines()
        n = sum(1 for ln in lines if cyr.search(ln))
        if n > 0:
            files.append((n, p))
            total_lines += n

files.sort(reverse=True)
print('files_with_cyrillic:', len(files))
print('cyrillic_lines_total:', total_lines)
print('--- top 30 ---')
for n, p in files[:30]:
    print(f'{n:6d}  {p}')
print('--- all files (for batching) ---')
for n, p in files:
    print(f'{n:6d}  {p}')
